code/modules/events/immovablerod.dm

```diff
@@ -101,9 +103,23 @@
 		qdel(src)
 		return
 
-	if(loc.density)
-		loc.ex_act(2)
-		if(prob(25))
+	if(loc && !istype(loc,/turf/space))
+		if(fuckMyShitUp)
+			loc.ex_act(1)
+		else if(loc.density)
+			if(istype(loc,/turf/simulated/wall/r_wall))
+				loc.ex_act(1)
```

Comment:
This is pretty bad, and we should probably move away from just using ex_act for everything because it's not really appropriate here. If you're going to typecheck for walls, you should use dismantle_wall(). But more generally, if you're going to break everything, you should just do a Destroy().